src/cli.rs

```diff
@@ -345,36 +345,47 @@ fn read_macaroon_from_file(path: PathBuf) -> Result<String, std::io::Error> {
     Ok(hex::encode(buffer))
 }
 
-fn read_cert_from_args(cert_pem: Option<String>, cert_path: Option<PathBuf>) -> ClientTlsConfig {
+fn read_cert_from_args(
+    cert_pem: Option<String>,
+    cert_path: Option<PathBuf>,
+) -> Result<ClientTlsConfig, String> {
     // Make sure both cert options are not set.
     if cert_path.is_some() && cert_pem.is_some() {
-        println!("ERROR: Only one of `cert_path` or `cert_pem` should be set.");
-        exit(1)
+        return Err("ERROR: Only one of `cert_path` or `cert_pem` should be set.".to_string());
     }
 
     let pem = match (&cert_pem, &cert_path) {
         (Some(pem), _) => pem.clone(),
-        (None, Some(cert_path)) => std::fs::read_to_string(cert_path).unwrap_or_else(|e| {
-            println!("ERROR reading cert: {e:?}");
-            exit(1)
-        }),
+        (None, Some(cert_path)) => std::fs::read_to_string(cert_path)
+            .map_err(|e| format!("ERROR reading cert: {:?}", e))?,
         (None, None) => {
             // If no cert pem string is provided, we'll look for the tls certificate in the
             // default location.
             let data_dir = home::home_dir()
                 .unwrap()
                 .join(DEFAULT_LNDK_DIR)
                 .join(DEFAULT_DATA_DIR);
-            std::fs::read_to_string(data_dir.join(TLS_CERT_FILENAME)).unwrap_or_else(|e| {
-                println!("ERROR reading cert: {e:?}");
-                exit(1)
-            })
+            std::fs::read_to_string(data_dir.join(TLS_CERT_FILENAME))
+                .map_err(|e| format!("ERROR reading cert: {:?}", e))?
         }
     };
     let cert = Certificate::from_pem(pem);
-    ClientTlsConfig::new()
+    Ok(ClientTlsConfig::new()
         .ca_certificate(cert)
-        .domain_name("localhost")
+        .domain_name("localhost"))
+}
+
+fn read_cert_from_args_or_exit(
+    cert_pem: Option<String>,
+    cert_path: Option<PathBuf>,
+) -> ClientTlsConfig {
+    match read_cert_from_args(cert_pem, cert_path) {
+        Ok(config) => config,
+        Err(err) => {
+            println!("{}", err);
+            exit(1);
```

Comment:
pre-existing, but i believe we can do something more idiomatic when exiting and tap into clap's errors for output. not needed in this PR.

Reply:
added an issue for this https://github.com/lndk-org/lndk/issues/202